lib/SIL/SILVerifier.cpp

```diff
@@ -2682,7 +2682,7 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
     require(
         !isa<WitnessMethodInst>(branchArg) &&
```

Comment:
@jckarter As I answered on https://github.com/apple/swift/pull/2499, IRGen currently crashes on any witness_methods passed as BB arguments. We discussed it this morning with @rjmccall and the conclusion was to introduce a stricter check in the verifier for the time being until the IRGen issue is solved. 

FYI, the fired assertion in IRGen is:

```
Assertion failed: (i == 0 && "tinyvector index out of range"), function operator[], file /Users/roman/src/s/llvm/include/llvm/ADT/TinyPtrVector.h, line 172.
0  swift                    0x000000010c2e1b8b llvm::sys::PrintStackTrace(llvm::raw_ostream&) + 43
1  swift                    0x000000010c2e0de6 llvm::sys::RunSignalHandlers() + 70
2  swift                    0x000000010c2e23af SignalHandler(int) + 575
3  libsystem_platform.dylib 0x00007fff94a5052a _sigtramp + 26
4  swift                    0x000000010ce50435 clang::vfs::getNextVirtualUniqueID()::UID + 73789
5  swift                    0x000000010c2e20c6 abort + 22
6  swift                    0x000000010c2e20a1 __assert_rtn + 81
7  swift                    0x00000001099882aa llvm::TinyPtrVector<llvm::PHINode*>::operator[](unsigned int) const + 58
8  swift                    0x00000001099864db addIncomingSILArgumentsToPHINodes((anonymous namespace)::IRGenSILFunction&, (anonymous namespace)::LoweredBB&, swift::OperandValueArrayRef) + 283
9  swift                    0x00000001099797e0 swift::SILVisitor<(anonymous namespace)::IRGenSILFunction, void>::visit(swift::ValueBase*) + 33552
10 swift                    0x000000010996d7eb (anonymous namespace)::IRGenSILFunction::emitSILFunction() + 7739
11 swift                    0x000000010996b5ca swift::irgen::IRGenModule::emitSILFunction(swift::SILFunction*) + 1178
12 swift                    0x00000001098a1c48 swift::irgen::IRGenerator::emitGlobalTopLevel() + 408
13 swift                    0x000000010994d32e performIRGeneration(swift::IRGenOptions&, swift::ModuleDecl*, swift::SILModule*, llvm::StringRef, llvm::LLVMContext&, swift::SourceFile*, unsigned int) + 974
14 swift                    0x000000010994c0c0 swift::performIRGeneration(swift::IRGenOptions&, swift::ModuleDecl*, swift::SILModule*, llvm::StringRef, llvm::LLVMContext&) + 656
15 swift                    0x000000010985a1dd performCompile(swift::CompilerInstance&, swift::CompilerInvocation&, llvm::ArrayRef<char const*>, int&, swift::FrontendObserver*) + 16701
16 swift                    0x0000000109855420 swift::performFrontend(llvm::ArrayRef<char const*>, char const*, void*, swift::FrontendObserver*) + 2928
17 swift                    0x000000010981ea4a main + 2986
18 libdyld.dylib            0x00007fff908d05ad start + 1
```

So, it looks like there is a mismatch in the number of arguments to PHI nodes. I'm not sure if it is related to dependencies on opened existentials.